### PR TITLE
change safe_logs test from IEEE_NORMAL to IEEE_FINITE

### PR DESCRIPTION
**Pull request by matthiasfabry**
addresses #510.

All tests on [Cannon](https://testhub.mesastar.org/change_safe_logs/commits/3529123) are passing, so I'd propose to implement this change into main. Although we'd probably want to test also on other systems too, since these low level routines could be machine/architecture/compiler dependent.

### CODE REVIEW

**Comment by fxt44** [comment]
i've started full test suite runs on intel and m1 macs.

File: math/public/math_lib_crmath.f90 [hunk 1]
```diff
@@ -172,7 +172,7 @@ elemental function safe_log10_ (x) result (log10_x)
     real(dp), intent(in) :: x
     real(dp)             :: log10_x
 
-    if (.NOT. IEEE_IS_NORMAL(x)) then
+    if (.NOT. IEEE_IS_FINITE(x)) then
 
        log10_x = 1.E-99_dp
```
Comment (evbauer) [inline math/public/math_lib_crmath.f90:177]: Do we want to return `1.E-99` here or `-99`?

File: math/public/math_lib_intrinsic.f90 [hunk 2]
```diff
@@ -171,7 +171,7 @@ elemental function safe_log_ (x) result (log_x)
     real(dp), intent(in) :: x
     real(dp)             :: log_x
 
-    if (.NOT. IEEE_IS_NORMAL(x)) then
+    if (.NOT. IEEE_IS_FINITE(x)) then
 
        log_x = 1.E-99_dp
```
Comment (evbauer) [inline math/public/math_lib_intrinsic.f90:176]: Do we want to return `1.E-99` here or `-99`?

**Comment by evbauer** [comment]
Or maybe we would want these cases to return some other large negative value (like `-101`) that specifically signals that they come from hitting this condition?

**Review by rhdtownsend — APPROVED** [review]

**Pull request merged**